pox/lib/addresses.py

```diff
@@ -199,7 +199,7 @@ def to_str (self, separator = ':', resolve_names  = False):
       # Don't even bother for local (though it should never match and OUI!)
       name = _eth_oui_to_name.get(self._value[:3])
       if name:
-        rest = separator.join('%02x' % (ord(x),) for x in self._value[3:])
+        rest = separator.join('%02x' % (ord(x),) for x in self._value[3:]) #TODO ERROR
```

Comment:
Offhand, this seems like it should be fine to me.  I think the problem is probably that there are cases where _value doesn't end up as a bytes object and is a string instead, which it shouldn't be.  This is probably because some of the code paths in the constructor were sloppy.  (On that note, there are constructor code paths that can be improved since raw inputs and hex inputs are now more easily distinguished via their type.)

Reply:
I updated all constructors to store values as bytes, in fact this is probably an outdated commit... `ord` is not needed anymore as bytes and integers are compatible